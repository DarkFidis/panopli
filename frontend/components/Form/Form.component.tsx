import React from "react";
import styles from "../../styles/Form.module.css";
import {FormProps} from "../../types/FormProps";
import { Formik } from "formik";

export const Form: React.FC<FormProps> = ({ submit, origin }) => (
  <div className={styles.formContainer}>
    <Formik
      onSubmit={submit}
      initialValues={{minDistance: 0, maxDistance: 10000}}
    >
      {({ values, handleBlur, handleChange, handleSubmit, isSubmitting}) => (
        <form onSubmit={handleSubmit}>
            <h3>Trouve ton resto</h3>
            <span>Distance minimale</span>
            <input
              type="number"
              name="minDistance"
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.minDistance}
            />
            <span>Distance maximale</span>
            <input
              type="number"
              name="maxDistance"
              onChange={handleChange}
              onBlur={handleBlur}
              value={values.maxDistance}
            />
            <br/>
            <button type="submit" color="info" disabled={!origin }>Send</button>
        </form>
      )}
    </Formik>
  </div>
)

export default Form
