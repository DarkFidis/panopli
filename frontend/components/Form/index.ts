import FormComponent, { Form } from './Form.component'
export { Form }
export default FormComponent
