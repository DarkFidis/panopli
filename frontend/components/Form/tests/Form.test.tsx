import * as React from 'react';

import { render, screen, waitFor } from '@testing-library/react';

import Form from '../Form.component'

describe('Form component unit test', () => {
  it('renders Form component', () => {
    // Given
    const origin = [48.85884, 2.3473]
    const submit = () => {}
    // When
    const { container } = render(<Form origin={origin} submit={submit} />)
    // Then
    expect(container).toBeInTheDocument();
  })
  it('should trigger submit prop when click on submit button', async () => {
    const origin = [48.85884, 2.3473]
    const submit = jest.fn()
    // When
    render(<Form origin={origin} submit={submit} />)
    // Then
    await waitFor(() => {
      const submitButton = screen.getByRole('button')
      submitButton.click()
      expect(submit).toHaveBeenCalled()
    })
  })
  it('button should be disabled given no origin prop', () => {
    const origin = null
    const submit = jest.fn()
    // When
    render(<Form origin={origin} submit={submit} />)
    // Then
    const submitButton = screen.getByRole('button')
    expect(submitButton).toHaveProperty('disabled', true)
  })
  it('should match snapshot', () => {
    // Given
    const origin = [48.85884, 2.3473]
    const submit = () => {}
    // When
    const { container } = render(<Form origin={origin} submit={submit} />)
    // Then
    expect(container).toMatchSnapshot()
  })
})
