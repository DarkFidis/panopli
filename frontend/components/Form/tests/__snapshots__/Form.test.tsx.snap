// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component unit test should match snapshot 1`] = `
<div>
  <div
    class="formContainer"
  >
    <form>
      <h3>
        Trouve ton resto
      </h3>
      <span>
        Distance minimale
      </span>
      <input
        name="minDistance"
        type="number"
        value="0"
      />
      <span>
        Distance maximale
      </span>
      <input
        name="maxDistance"
        type="number"
        value="1000"
      />
      <br />
      <button
        color="info"
        type="submit"
      >
        Send
      </button>
    </form>
  </div>
</div>
`;
